### PR TITLE
feat: enable shell for --oci mode, from sylabs 1133 & 1143 & 1145

### PR DESCRIPTION
**Pull request by edytuk**
This pulls in sylabs PRs

- sylabs/singularity# 1133
- sylabs/singularity# 1143
- sylabs/singularity# 1145
 which fixed
- sylabs/singularity# 1025
- sylabs/singularity# 1041
- sylabs/singularity# 1042
- sylabs/singularity# 1044

The original PR descriptions were:
> Enable `singularity shell --oci ...` with behavior matching native runtime, i.e.
> 
> * Run shell set with SINGULARITY_SHELL or --shell 
> * If not set, try /bin/bash --norc 
> * If not available, use /bin/sh

> Tidy up the rpm spec file.
> * Remove the manual handling of build root, GOPATH etc. Not needed as we are using go modules now.
> * Remove redundant explicit deps.
> * Fix crun -> runc dep for EL7, SLES.
> * Use rpm make_xxx macros instead of direct make calls.
> * Ensure all directories created are owned by package.

> If `crun` is not available, require `runc`.

### CODE REVIEW

File: internal/pkg/runtime/launcher/native/launcher_linux.go [hunk 1]
```diff
@@ -26,7 +26,8 @@ import (
 	"github.com/apptainer/apptainer/internal/pkg/buildcfg"
 	"github.com/apptainer/apptainer/internal/pkg/cgroups"
 	"github.com/apptainer/apptainer/internal/pkg/checkpoint/dmtcp"
-	"github.com/apptainer/apptainer/internal/pkg/fakeroot"
+	fakefake "github.com/apptainer/apptainer/internal/pkg/fakefake"
```
Comment (DrDaveD) [inline internal/pkg/runtime/launcher/native/launcher_linux.go:29]: There's no need for an alias here.

File: internal/pkg/runtime/launcher/native/launcher_linux.go [hunk 2]
```diff
@@ -26,7 +26,8 @@ import (
...
+	fakerootutil "github.com/apptainer/apptainer/internal/pkg/fakeroot"
```
Comment (DrDaveD) [inline internal/pkg/runtime/launcher/native/launcher_linux.go:30]: Why is this alias necessary?  It would be better to just use fakeroot

**Review by DrDaveD — CHANGES_REQUESTED** [review]

File: internal/pkg/runtime/engine/apptainer/process_linux.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ import (
 	"unsafe"
 
 	"github.com/apptainer/apptainer/internal/pkg/checkpoint/dmtcp"
-	"github.com/apptainer/apptainer/internal/pkg/fakeroot"
+	fakeroot "github.com/apptainer/apptainer/internal/pkg/fakefake"
```
[the remark this answers is not in the document]
Comment (edytuk) [reply internal/pkg/runtime/engine/apptainer/process_linux.go:35]: There was error without alias
`internal/pkg/runtime/engine/apptainer/process_linux.go:35: File is not `goimports`-ed (goimports)`

File: internal/pkg/runtime/launcher/native/launcher_linux.go (hunk 1 above)
Comment (edytuk) [reply internal/pkg/runtime/launcher/native/launcher_linux.go:29]: There were errors without alias
```
Error: ../internal/pkg/runtime/launcher/native/launcher_linux.go:30:2: fakeroot redeclared in this block
Error: 	../internal/pkg/runtime/launcher/native/launcher_linux.go:29:2: other declaration of fakeroot
Error: ../internal/pkg/runtime/launcher/native/launcher_linux.go:30:2: imported and not used: "github.com/apptainer/apptainer/internal/pkg/fakeroot"
Error: ../internal/pkg/runtime/launcher/native/launcher_linux.go:128:40: undefined: fakeroot.IsUIDMapped
Error: ../internal/pkg/runtime/launcher/native/launcher_linux.go:129:80: undefined: fakeroot.SubUIDFile
Error: ../internal/pkg/runtime/launcher/native/launcher_linux.go:148:124: undefined: fakeroot.SubUIDFile
```

Comment (DrDaveD) [reply internal/pkg/runtime/launcher/native/launcher_linux.go:29]: I mean for the functions that are in the fakefake file.

File: internal/pkg/runtime/engine/apptainer/process_linux.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (DrDaveD) [reply internal/pkg/runtime/engine/apptainer/process_linux.go:35]: There must have been a sorting error or something like that.  The easiest fix for that is to run `gofumpt -w` on the file.

[the remark this answers is not in the document]
Comment (DrDaveD) [reply internal/pkg/runtime/engine/apptainer/process_linux.go:35]: gofumpt is upward compatible from goimports, so we use gofumpt.

**Comment by edytuk** [comment]
I made changes related with fakefake file and added 1145 PR, you can review again

**Review by DrDaveD — APPROVED** [review]

File: INSTALL.md [hunk 4]
```diff
@@ -27,9 +27,14 @@ sudo apt-get install -y \
...
     fakeroot \
     cryptsetup \
-    curl wget git
+    curl wget git \
+    conmon crun
 ```
 
+_Note - leave out `conmon` and `crun`, which will prevent the `--oci` mode from
+working on Ubuntu 18.04 or Debian 10. `fuse-overlayfs` doesn't exist on
+Ubuntu 18.04 or Debian 10._
```
Comment (DrDaveD) [inline INSTALL.md:36]: Oh, I approved before reading this wording.  It is hard to follow.  I suggest instead:
```
_Note_: on Ubuntu 18.04 or Debian 10 leave out `conmon`, `crun`, and `fuse-overlayfs` because they are not available, or install them from another source.  Leaving out the first two will prevent the `--oci` option from working and leaving out the third will prevent `--overlay` and `--writable-tmpfs` options from working.
```
